python_modules/dagster-graphql/dagster_graphql/implementation/loader.py

```diff
@@ -277,3 +284,111 @@ def get_latest_materialization_for_asset_key(self, asset_key: AssetKey) -> Event
     def _fetch(self):
         self._fetched = True
         self._materializations = self._instance.get_latest_materialization_events(self._asset_keys)
+
+
+class CrossRepoAssetDependencyLoader:
+    """
```

Comment:
A thought, which you can feel free to ignore: would it be easier to have this class be responsible for _all_ dependency loading, instead of just cross-repo dependency loading?

Reply:
Is it for readability purposes that this class would contain all dependency loading?

This class needs to differentiate between assets using their locations (repo location name and repository name), and this information doesn't exist on the asset node object itself, it would need to be passed to this class.

I think that it would be extraneous to pass in this information for dependencies (where all the necessary information exists on the ExternalAssetNode). I think I prefer isolating this class purely for cross-repo `DependedBy` loading.

Reply:
Is it for readability purposes that this class would contain all dependency loading?

This class needs to differentiate between assets using their locations (repo location name and repository name), and this information doesn't exist on the asset node object itself, it would need to be passed to this class.

I think that it would be extraneous to pass in this information for dependencies (where all the necessary information exists on the ExternalAssetNode). I think I prefer isolating this class purely for cross-repo `DependedBy` loading.